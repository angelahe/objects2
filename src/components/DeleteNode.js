import React from 'react';
import '../styles/Styles140.css'
import closebtn from '../images/close_FFFFFF.png';

const DeleteNode = (props) => {

  function handleCloseClick() {
    props.closeClicked("delete");
  }

  function handleDeleteClick() {
    console.log("in handleUpdateClick");
    props.deleteSubject(props.node.subject);
  }

  const subject = props.node.subject;
  const amount = props.node.amount;

  return(
    <div>
      <div className = "ItemBox AppHeader">
        <span className = "AddItem">Delete Account</span>
        <button className="AppBtn" onClick={handleCloseClick}>
          <img className="btnImg" src={closebtn} alt="Close"/>
        </button>
      </div>
      <div className = "ItemBox">
        <span className = "DetailText">Subject:</span>
        <span className = "DetailText">{subject}</span><br/>
        <span className = "DetailText">Amount: </span>
        <span className = "DetailText">{amount}</span><br/>
        <button className="btnText" onClick={handleDeleteClick}>Delete</button>
      </div>
    </div>
  )

}
export default DeleteNode;