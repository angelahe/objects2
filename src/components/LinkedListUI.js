import React from "react";
import linkedlist from "./linkedlist";
import AddNode from "./AddNode";
import addbtn from '../images/add_FFFFFF.png';
import deletebtn from '../images/delete_FFFFFF.png';
import '../styles/Styles140.css';
import editbtn from "../images/edit_FFFFFF.png";

class LinkedListUI extends React.Component {

  constructor () {
    super()
    this.state = {
      SubjectList: new linkedlist(),
      addShow: false,
      deleteShow: false,
      subjectAsc: true,
      amountAsc: true,
      currentNode: null
    }

  }

  onBtnCloseClick = (closeType) => {
    if (closeType === "create") this.setState ({addShow: false});
    if (closeType === "delete") this.setState ({deleteShow: false});
  }

  onAddClick = () => {
    this.setState ({addShow: true});
  }

  onDeleteClick = () => {
    this.setState ({deleteShow: true});
  }

  onSubjectSort = () => {
    this.setState ({subjectAsc: !this.state.subjectAsc});
  }

  onSubjectSort = () => {
    this.setState ({amountAsc: !this.state.amountAsc});
  }

  onSubjectCreate = (subject, amount) => {

    if (this.state.SubjectList.length === 0) {
      this.state.SubjectList.addNode(subject, amount);
      const currentNode = this.state.SubjectList.head;
      console.log("head is", currentNode);

      this.setState({
        SubjectList: this.state.SubjectList,
        currentNode: currentNode
      });
    }
    else {
      console.log("current node is", this.state.currentNode);
      const index = this.state.SubjectList.indexOf(this.state.currentNode.subject);
      this.state.SubjectList.insertNodeAt(index, subject, amount);
      this.setState({
        SubjectList: this.state.SubjectList,
        currentNode: this.state.SubjectList.nodeAt(index)
      });
    }
    this.setState({
      addShow: false
    })
    console.log(this.state);

  }

  render() {

    return(
      <div>
        <h1>Hello from LinkedList</h1>
        <div className = "AppContainer">
          <div className = "AppPanel">
            <div className = "ItemBox AppHeader">
              <span className = "AddItem">Add Subject</span>
              <button className="AppBtn" onClick={this.onAddClick}>
                <img className="btnImg" src={addbtn} alt="Add"/>
              </button>
            </div>
            <div className = "AppList">
              {(this.state.currentNode !== null)
              ? <div>
                  <h2>{this.state.currentNode.subject}</h2>
                  <h2>{this.state.currentNode.amount}</h2>
                  <button className="AppBtn" onClick={this.onAddClick}>
                  <img className="btnImg" src={editbtn} alt="Add" onClick={this.onAddClick}/>
                  </button>
                  <button className="AppBtn Delete" onClick={this.onDeleteClick}>>
                  <img className="btnImg" src={deletebtn} alt="Add" onClick={this.onDeleteClick}/>
                  </button>
                </div>
              : null}

            </div>
          </div>
          <div className = "AppPanel">
            {(this.state.addShow)
            ? <AddNode createClicked = {this.onSubjectCreate}
                         closeClicked = {this.onBtnCloseClick}
              />
            : null
            }
            {(this.state.deleteShow)
            ? <h1>show delete dialog</h1>
            :null
            }
          </div>
        </div>
      </div>
    )
  }

}

export default LinkedListUI;