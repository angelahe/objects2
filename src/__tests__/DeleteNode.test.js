import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import DeleteNode from "../components/DeleteNode";
import LinkedListUI from "../components/LinkedListUI";
import renderer from "react-test-renderer";

const props = new LinkedListUI();
const node = {subject: "Kitten",
              amount: 100,
              next: null };

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<DeleteNode node = {node}
                              deleteSubject = {props.onSubjectDelete}
                              closeClicked = {props.onBtnCloseClick}
  />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  shallow(<DeleteNode node = {node}
                      deleteSubject = {props.onSubjectDelete}
                      closeClicked = {props.onBtnCloseClick}
  />);
});

test('Starter renders correctly', () => {
  const tree = renderer.create(<DeleteNode node = {node}
                                           deleteSubject = {props.onSubjectDelete}
                                           closeClicked = {props.onBtnCloseClick}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});