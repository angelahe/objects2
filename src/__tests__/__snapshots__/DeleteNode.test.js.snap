// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Starter renders correctly 1`] = `
<div>
  <div
    className="ItemBox AppHeader"
  >
    <span
      className="AddItem"
    >
      Delete Account
    </span>
    <button
      className="AppBtn"
      onClick={[Function]}
    >
      <img
        alt="Close"
        className="btnImg"
        src="close_FFFFFF.png"
      />
    </button>
  </div>
  <div
    className="ItemBox"
  >
    <span
      className="DetailText"
    >
      Subject:
    </span>
    <span
      className="DetailText"
    >
      Kitten
    </span>
    <br />
    <span
      className="DetailText"
    >
      Amount: 
    </span>
    <span
      className="DetailText"
    >
      100
    </span>
    <br />
    <button
      className="btnText"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
